x-pack/plugins/maps/public/connected_components/mb_map/index.ts

```diff
@@ -42,6 +43,9 @@ function mapStateToProps(state: MapStoreState) {
     inspectorAdapters: getInspectorAdapters(state),
     scrollZoom: getScrollZoom(state),
     isFullScreen: getIsFullScreen(state),
+    featureModeActive:
```

Comment:
wrt comment of using a new store-object `EditState`. these evaluations for featureModeActive and filterMpdeAvitve will be a little different. Basically, `featureMode` (=EditState) and `filterMode` (=DrawState) would be made explicit in the store as well.

Reply:
I made the requested revisions to `EditState` and `DrawState` but they're different from the draw mode. These two states just track information about the shapes drawn. We're still tracking `DRAW_MODE` as a separate ui toggle via redux, which I think is easier and clearly differentiates which MB draw tools instance we're using at any given point